synphot/wrapper.py

```diff
@@ -0,0 +1,153 @@
+                raise exceptions.BandpassError("bandpass should be "
+                                               "a list with 2 x n "
+                                               "dimensions.")
+            # order of waves/transmittance given by bandpass doesn't matter:
+            try:
+                self.bandpass = _call_SpecElement_4list(self._bandpass[0],
+                                                        self._bandpass[1])
+            except u.UnitsError:
+                try:
+                    self.bandpass = _call_SpecElement_4list(self._bandpass[1],
+                                                            self._bandpass[0])
+                except u.UnitsError:
+                    print("rows of bandpass list must be of type "
+                          "astropy.units.quantity.Quantity")
+                
+        elif type(self._bandpass) == str:
+            self.bandpass = SpectralElement.from_filter(self._bandpass)
+
+        elif type(self._bandpass) == SpectralElement:
+            self.bandpass = self._bandpass
+
+        else:
+            raise exceptions.BandpassError("bandpass must be of type str, "
+                "`~synphot.spectrum.SpectralElement`, or "
+                "list of shape(2, n) containing "
+                "`~astropy.units.quantity.Quantity` values")
+
+        self.throughput = self.bandpass
+
+    def _include_qe(self):
+        if len(self._qe) != 2:
+            raise exceptions.BandpassError("quantum_efficiency should be "
+                                           "a list with 2 x n "
+                                           "dimensions.")
+        try:
+            self.quantum_efficiency = _call_SpecElement_4list(self._qe[0], self._qe[1])
+        except u.UnitsError:
+            try:
+                self.quantum_efficiency = _call_SpecElement_4list(self._qe[1], self._qe[0])
+            except u.UnitsError:
+                print("rows of quantum_efficiency list must be of type "
+                      "astropy.units.quantity.Quantity")
+
+        self.throughput *= self.quantum_efficiency
+
+    def _include_atmosphere(self):
+        if len(self._at) != 2:
+            raise exceptions.BandpassError("atmospheric_transmission "
+                                           "should be "
+                                           "a list with 2 x n "
+                                           "dimensions.")
+        try:
+            self.atmospheric_transmission = _call_SpecElement_4list(self._at[0], self._at[1])
+        except u.UnitsError:
+            try:
+                self.atmospheric_transmission = _call_SpecElement_4list(self._at[1], self._at[0])
+            except u.UnitsError:
+                print("rows of atmospheric_transmission list must be of type "
+                      "astropy.units.quantity.Quantity")
+
+        self.throughput *= self.atmospheric_transmission
+
+@u.quantity_input(waves=u.angstrom, frac=u.Unit(''))
```

Comment:
I don't think a private method needs a decorator to enforce that you're using it correctly. Ideally the user will never know this function exists, so they shouldn't need forceful reminders to get their units correct).

Reply:
Ah yeah, I put that decorator where it is because there doesn't seem to be a better place to put it. Passing these parameters in as lists is not the default option, so I couldn't put it before `__init__()`, but I do still want to enforce that the user's custom models are in the correct units.